### DISCUSSION TITLE
Is there a way to create clear all filters button?

### DISCUSSION THREAD

**Discussion by ashatat** [KevinVandy/material-react-table#455] [Q&A] [answered]
Hi,  I need a button to clear all column filters at once, I can manage the `columnFilters` state and clear it when clicking on the button.
what I am struggling with is that I want to add the button as an Icon next to show/hide filters 
Any suggestions please :pray:?

![55555555](https://user-images.githubusercontent.com/18149438/229781657-d230969b-7a1a-452a-9b69-4d473a34552c.png)

**Comment by KevinVandy** [comment]
There is a table instance resetColumnFilters api

<img width="576" alt="image" src="https://user-images.githubusercontent.com/28243511/229837893-ae8d959c-70bc-4d61-b660-9625ba224c74.png">

Combine that knowledge with the custom internal[ toolbar buttons example](https://www.material-react-table.com/docs/guides/customize-toolbars#customize-built-in-internal-toolbar-button-area)

**Reply by ashatat** [reply]
hi @KevinVandy  thanks for replying
I am struggling to find any doc on how to use resetColumnFilters?? should I just pass a boolean for it to work? or is it a function?

been trying to make it work with no luck

**Comment by ashatat** [comment]
For anyone in the future looking for this.
This  is a workaround to make this work, (it's a shame that I couldn't use `resetColumnFilters` since no documentation to show how it suppose to work)

```js
function CustomTable({ data, columns, isLoading, state, ...rest }) {
	const [columnFilters, setColumnFilters] = useState([]);
	const [showColumnFilters, setShowColumnFilters] = useState(false);

	return (
		<MaterialReactTable
			columns={columns}
			data={data}
			state={{ isLoading, columnFilters, showColumnFilters, ...state }}
			onColumnFiltersChange={setColumnFilters}
			onShowColumnFiltersChange={() => setShowColumnFilters(v => !v)}
			//customize built-in buttons in the top-right of top toolbar
			renderToolbarInternalActions={({ table }) => (
				<Box>
					{!!columnFilters.length && (
						<Tooltip title="Reset filters" arrow>
							<IconButton
								onClick={() => {
									setShowColumnFilters(false);
									setColumnFilters([]);
								}}
							>
								<FilterAltOffOutlinedIcon />
							</IconButton>
						</Tooltip>
					)}
					{/* Built-in buttons*/}
					<MrtToggleFiltersButton table={table} />
					<MrtShowHideColumnsButton table={table} />
					<MrtToggleDensePaddingButton table={table} />
					<MrtFullScreenToggleButton table={table} />
				</Box>
			)}
			{...rest}
		/>
	);
}
```

**Reply by joey-kwl** [reply, marked as answer]
You can use `resetColumnFilters` by calling `table.resetColumnFilters()`
